### PR TITLE
Product SKU input scanner release preparation

### PR DESCRIPTION
**Pull request by jaclync**
Completes the first feature of #2407 

### Description

We are ready to release the product SKU input scanner as the first feature of barcode scanner, and this PR contains a few updates before the release:

- Created a new feature flag `productSKUInputScanner` that always returns `true`, which is used for product SKU input scanner CTA. Since the more complicated inventory scanner (CTA on the Products tab) is still WIP, we want to release two features separately with two feature flags
- Added a feature switch for the feature (ref p91TBi-6Sv-p2#comment-7889)
- For both barcode scanner CTAs, added a check to only show the CTAs when camera is available on the device `UIImagePickerController.isSourceTypeAvailable(.camera)`
- Updated camera permission purpose string in `Info.plist` to include product SKU scanner

### Testing instructions

Let's simulate release configuration by testing with a device and manually disabling the `barcodeScanner` feature flag:

- In `DefaultFeatureFlagService`, return `false` for `barcodeScanner` to simulate release build (while `productSKUInputScanner` feature is always enabled)
- Launch the app on a device
- Go to the products tab --> the scan CTA should not be visible (behind `barcodeScanner` feature flag)
- Tap on a product whose SKU is editable
- Tap on the inventory/SKU row --> the scan CTA should not be visible because the feature switch is disabled by default
- Leave the product form and go to the My Store tab
- Go to Settings > Experimental Features --> there should be a section for Product SKU Scanner with a switch that is disabled by default
- Turn on the feature switch in the Product SKU Scanner section
- Leave the settings and go to the products tab
- Tap on a product whose SKU is editable
- Tap on the inventory/SKU row --> the scan CTA should be visible now
- Tap on the scan CTA --> it should navigate to the SKU input scanner

#### Other scenarios

@jaclync tests the following:

- [x] The feature switch remains the same state after switching stores
- [x] The feature switch remains the same state after logging out and in

### Screenshots

iOS camera permission alert | experimental features | scan CTA when feature switch is enabled
-- | -- | --
 |  | 

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jaclync** [comment]
@AliSoftware I updated the camera permission in `Info.plist` here for the product SKU input scanner release 🙂 please lemme know if there are any other changes I should make!

**Comment by AliSoftware** [comment]
@jaclync I didn't expect this to land so soon, my part of the tooling to address this is not ready to handle this yet after all 😅 

So I think that, especially since in WCiOS those keys were apparently never translated so far in the first place (as there's no `InfoPlist.strings` in the project at all to handle localization of those, it seems), you can land this PR as is for now.

Once I've implemented the plan (ref: paaHJt-2Px-p2) for handling localization of those keys in WPiOS and validated that it works there, I'll spend some time to set up WCiOS similarly so that it handles localization of those here as well… but that will probably only land in WCiOS around mid-January, so don't let this be a blocker for your PR.

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift [hunk 1]
```diff
@@ -284,14 +284,24 @@ private extension ProductInventorySettingsViewController {
...
+                return cell.accessoryView = nil
+            }
+
             let button = UIButton(type: .detailDisclosure)
             button.applyIconButtonStyle(icon: .scanImage)
-            button.addTarget(self, action: #selector(scanSKUButtonTapped), for: .touchUpInside)
-
+            button.on(.touchUpInside) { [weak self] _ in
+                self?.scanSKUButtonTapped()
+            }
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift]: We can also use the native method `addAction`:
```suggestion
            button.addAction(UIAction { [weak self] _ in
                self?.scanSKUButtonTapped()
            }, for: .touchUpInside)
```

Also, I think we should remove the `@objc` attribute for the method `scanSKUButtonTapped` 😄

**Review by itsmeichigo — APPROVED** [review]
Code looks good and works as expected! 👏

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift]: Great suggestion! TIL [iOS 14+ `UIControl.addAction` ](https://developer.apple.com/documentation/uikit/uicontrol/3600490-addaction) 😮 updated in https://github.com/woocommerce/woocommerce-ios/pull/5695/commits/452c9cc4e2496685d9950d659170bec9efa77478

**Pull request merged**